contracts/modules/cover/Cover.sol

```diff
@@ -0,0 +1,773 @@
+  ) external payable onlyMember whenNotPaused {
+
+    CoverData memory cover = _coverData[coverId];
+    uint lastCoverSegmentIndex = _coverSegments[coverId].length - 1;
+    CoverSegment memory lastCoverSegment = coverSegments(coverId, lastCoverSegmentIndex);
+
+    require(ICoverNFT(coverNFT).isApprovedOrOwner(msg.sender, coverId), "Cover: Only owner or approved can edit");
+    require(lastCoverSegment.start + lastCoverSegment.period > block.timestamp, "Cover: cover expired");
+    require(buyCoverParams.period < MAX_COVER_PERIOD, "Cover: Cover period is too long");
+    require(buyCoverParams.commissionRatio <= MAX_COMMISSION_RATIO, "Cover: Commission rate is too high");
+
+
+    // Override cover specific parameters
+    buyCoverParams.payoutAsset = cover.payoutAsset;
+    buyCoverParams.productId = cover.productId;
+
+    uint refundInCoverAsset = 0;
+
+    if (!lastCoverSegment.expired) {
+      uint32 remainingPeriod = lastCoverSegment.start + lastCoverSegment.period - uint32(block.timestamp);
+
+      PoolAllocation[] storage originalPoolAllocations = coverSegmentAllocations[coverId][lastCoverSegmentIndex];
+
+      {
+        uint originalPoolAllocationsCount = originalPoolAllocations.length;
+
+        // rollback previous cover
+        for (uint i = 0; i < originalPoolAllocationsCount; i++) {
+          stakingPool(originalPoolAllocations[i].poolId).deallocateStake(
+            cover.productId,
+            lastCoverSegment.start,
+            lastCoverSegment.period,
+            originalPoolAllocations[i].coverAmountInNXM,
+            originalPoolAllocations[i].premiumInNXM,
+            lastCoverSegment.globalRewardsRatio
+          );
+
+          originalPoolAllocations[i].premiumInNXM *= (lastCoverSegment.period - remainingPeriod) / lastCoverSegment.period;
+
+          // compute NXM rewards deallocated
+          uint deallocatedRewardsInNXM = originalPoolAllocations[i].premiumInNXM
+          * remainingPeriod / lastCoverSegment.period
+          * lastCoverSegment.globalRewardsRatio / REWARD_DENOMINATOR;
+          tokenController().burnStakingPoolNXMRewards(deallocatedRewardsInNXM, originalPoolAllocations[i].poolId);
+        }
+      }
+
+      refundInCoverAsset = lastCoverSegment.priceRatio
+        * lastCoverSegment.amount
+        / PRICE_DENOMINATOR
+        * remainingPeriod
+        / MAX_COVER_PERIOD;
+
+      // edit cover so it ends at the current block
+      lastCoverSegment.period = lastCoverSegment.period - remainingPeriod;
+    }
+
+    uint totalPremiumInNXM = _buyCover(buyCoverParams, coverId, poolAllocations);
+
+    handlePaymentAndRefund(buyCoverParams, totalPremiumInNXM, refundInCoverAsset);
+
+    // update total cover amount for asset by decreasing the previous amount and adding the new amount
+    if (coverAmountTrackingEnabled) {
+       totalActiveCoverInAsset[cover.payoutAsset] =
+        totalActiveCoverInAsset[cover.payoutAsset] - lastCoverSegment.amount + buyCoverParams.amount;
```

Comment:
Should add the actual covered amount not the requested one (`buyCoverParams.amount` is requested, not covered)

Reply:
done